### ISSUE TITLE
A Proper Solution To Documentation/FAQ

### ISSUE THREAD

**Issue by Brittank88** [absoIute/Mega-Hack-Pro-Future#447]
[BookStack](https://www.bookstackapp.com/) is free, open-source, and self-hostable. It's already used in existing projects such as [KubeJS](https://mods.latvian.dev/books/kubejs), and we could use it to provide clear and detailed documentation on MHv5/6/7 (one book in the "Mega Hack" shelf per major version).

Within each book, we can have a chapter dedicated to a grouping of hacks, with one or more pages dedicated to describing the features, limitations, usages of each specific hack.

This brings us closer to the goal of everything being contained within Absolute's own server ecosystem and provides the general public with a proper way to learn everything about how Mega Hack works and what it offers. Only the other day a friend told me they thought MHv6/7 was just "MHv5 but with hitbox visualisation". This would help alleviate that.

The documentation could be linked directly from Mega Hack, as well as from Absolute's website.

I understand that [A mod listing](https://absolllute.com/store/mega_hack_mods) exists on Absolute's website, however, this doesn't feel very clean, nor very detailed. It lacks visual documentation, potential use cases, etc. and is also inaccurate as it is the MHv6 mods listing yet is linked to from the MHv7 page. It could be so much better! 😋

**Comment by Brittank88** [comment]
If this is accepted, I will edit the initial issue to be categorised with headings and include a checklist to track the progress of documentation writing for v1 of the docs.

Issues related to documentation following v1 can be tied into the MH version milestones.

**Comment by MainConMan** [comment]
> Only the other day a friend told me they thought MHv6/7 was just "MHv5 but with hitbox visualisation".

Clearly, they have not watched the Mega Hack v7 Pro youtube video.

Joking aside, I quickly read over the website, and it seems like a great way to accomplish this. Good idea!

**Comment by Nicko2102** [comment]
> Within each book, we can have a chapter dedicated to a grouping of hacks, with one or more pages dedicated to describing the features, limitations, usages of each specific hack.

If you want, I could help with the grouping and explanations for v6/v7

**Comment by MainConMan** [comment]
I will also try to help in any way I can! But I've never used BookStack before, so I probably won't be very helpful tbh.

**Comment by Nicko2102** [comment]
> I will also try to help in any way I can! But I've never used BookStack before, so I probably won't be very helpful tbh.

Me neither, but i could try to format it in such a way that can easily be moved to bookstack without issue

**Comment by MainConMan** [comment]
> > I will also try to help in any way I can! But I've never used BookStack before, so I probably won't be very helpful tbh.
> 
> Me neither, but i could try to format it in such a way that can easily be moved to bookstack without issue.

Ohh, I see what you're saying, maybe we could work together on that (if you're ok with that of course). Seems like a big project.

> Also, did github crash for anyone else or was that just me?

Idk, I was at lunch. I didn't notice anything.

**Comment by Nicko2102** [comment]
> Ohh, I see what you're saying, maybe we could work together on that (if you're ok with that of course). Seems like a big project.

sounds good to me :)

**Comment by MainConMan** [comment]
> > Ohh, I see what you're saying, maybe we could work together on that (if you're ok with that of course). Seems like a big project.
> 
> sounds good to me :)

Cool! :D Add me on discord, we should probably discuss it there so we don't fill up this issue too much. MainConMan#7911

**Comment by absoIute** [comment]
Definitely will look at something like that or similar, need to upgrade the mod list & changelog!!

**Comment by Brittank88** [comment]
> > > Ohh, I see what you're saying, maybe we could work together on that (if you're ok with that of course). Seems like a big project.
> > 
> > 
> > sounds good to me :)
> 
> Cool! :D Add me on discord, we should probably talk about it there so we don't fill up this issue too much. MainConMan#7911

Definitely add me at Brittank88#7106 too - Maybe we can form a group chat to work on this?

**Comment by Brittank88** [comment]
> Definitely will look at something like that or similar, need to upgrade the mod list & changelog!!

If you use BookStack you won't even need a mod list on your website, you can just link to the book for the corresponding version of MH!

In terms of a changelog, in #448 I proposed some ways of dealing with that which may be useful.

**Comment by defaultacc30** [comment]
> [BookStack](https://www.bookstackapp.com/) is free, open-source, and self-hostable. It's already used in existing projects such as [KubeJS](https://mods.latvian.dev/books/kubejs), and we could use it to provide clear and detailed documentation on MHv5/6/7 (one book in the "Mega Hack" shelf per major version).
> 
> Within each book, we can have a chapter dedicated to a grouping of hacks, with one or more pages dedicated to describing the features, limitations, usages of each specific hack.
> 
> This brings us closer to the goal of everything being contained within Absolute's own server ecosystem and provides the general public with a proper way to learn everything about how Mega Hack works and what it offers. Only the other day a friend told me they thought MHv6/7 was just "MHv5 but with hitbox visualisation". This would help alleviate that.
> 
> The documentation could be linked directly from Mega Hack, as well as from Absolute's website.
> 
> I understand that [A mod listing](https://absolllute.com/store/mega_hack_mods) exists on Absolute's website, however, this doesn't feel very clean, nor very detailed. It lacks visual documentation, potential use cases, etc. and is also inaccurate as it is the MHv6 mods listing yet is linked to from the MHv7 page. It could be so much better! 😋

i like the idea but bookstack isn't used that much so people might not click it

**Comment by Nicko2102** [comment]
> i like the idea but bookstack isn't used that much so people might not click it

I don't think it matters how well-known it is as long as there's a way to access it from Absolute's website

**Comment by Nicko2102** [comment]
thats what i was trying to say but couldnt find the words lol. Like nobody will see the link to the changelog but decide not to click it because they dont know what bookstack is.
